pandas/core/reshape/melt.py

```diff
@@ -134,7 +137,9 @@ def melt(
 
     mcolumns = id_vars + var_name + [value_name]
 
-    if frame.shape[1] > 0:
+    if frame.shape[1] > 0 and not any(
+        isinstance(dt, DatetimeTZDtype) for dt in frame.dtypes.values
```

Comment:
This may need to be more generic like https://github.com/pandas-dev/pandas/blob/e86ed377639948c64c429059127bcf5b359ab6be/pandas/core/reshape/melt.py#L125

Reply:
I ended up importing the types like so since pandas doesn't have a `dtype` method like numpy.  Lmk if you had something else in mind.
```Python
pd_dtypes = (
        CategoricalDtype,
        DatetimeTZDtype,
        ExtensionDtype,
        IntervalDtype,
        PeriodDtype,
        SparseDtype,
    )

if frame.shape[1] > 0 and not any(
        isinstance(dt, pd_dtypes) for dt in frame.dtypes.values
    )
```